test-data/unit/check-type-aliases.test

```diff
@@ -770,3 +770,71 @@ f(string, string)
 
 [builtins fixtures/tuple.pyi]
 [typing fixtures/typing-medium.pyi]
+
+[case testShowTypeAliasInErrorMessage]
+from typing import Tuple
+
+Point = Tuple[float, ...]
+
+def dist(q: Point) -> None:
+    pass
+
+dist('abc') # E: Argument 1 to "dist" has incompatible type "str"; expected "Point"
+
+[builtins fixtures/tuple.pyi]
+
+# See https://github.com/python/mypy/issues/2968#issuecomment-1133768276
+# We need to disable the no_args optimization to get this to work
```

Comment:
This will have bad performance impact while we still have `Text = str` and similar "plain" aliases.

Reply:
That's definitely a good point, but I'm not really sure I like the idea of mypy inconsistently showing alias names in error messages based on an implementation detail.

Either way, this PR doesn't actually change that behavior, and leaving the xfail test here seems like a good reminder that we need to discuss whether the performance benefits or consistency is more important (probably in the original issue).